src/app/api/v1/premium/route.ts

```diff
@@ -10,19 +10,12 @@ export async function GET(req: NextRequest) {
     // NOTE: the email address passed here cannot
     // be trusted, we must also check that the
     // `subscriptions` claim on the FxA token contains
-    // "monitor".
+    // "monitor", or call the auth server API to confirm.
 
     const { searchParams } = new URL(req.url);
     const _email = searchParams.get("email");
 
-    // TODO: The user either needs to be signed in again, or the
-    // JWT from FxA refreshed, in order to read the latest
-    // subscriptions.
-    //
-    // For now, redirect to the built-in sign-out, which will then
-    // send the user to the dashboard.
-
-    return NextResponse.redirect(`${process.env.SERVER_URL}/api/auth/signout?callbackUrl=/user/breaches`, 302);
```

Comment:
Hm, we could actually move the subscription check here and add it to the session if it doesn't already exist. In the meantime I'll get to the bottom of why the `subscription` claim doesn't seem to exist on the dev server - it is using a different client ID than we use for local development which could be why.

Reply:
I took a shot at this but if I `getSession()` and then modify it, I don't see my modification later when the page is rendered and I'm not sure why... I think it's not too important yet, there's only one place we currently check for subscription status, but I think checking in the callback from subplat and adding it to the session is probably a better place.